gooddata-server-oauth2-autoconfigure/src/main/kotlin/BearerTokenReactiveAuthenticationManagerResolver.kt

```diff
@@ -108,7 +138,79 @@ private class JwtAuthenticationManager(
                         withState("error")
                         withAuthenticationMethod(AUTH_METHOD)
                     }
-                    parseJwtException(ex, jwtToken)
+
+                    when (ex.cause?.cause) {
```

Comment:
I would suggest to extract the audit logging processing into a separate method to avoid huge methods.